File: app/src/main/java/in/completecourse/adapter/SliderAdapter.java
package in.completecourse.adapter;

import android.app.Activity;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import in.completecourse.R;
import in.completecourse.model.Update;

public class SliderAdapter extends PagerAdapter {

    private final Activity mContext;
    private ArrayList<Update> mUpdateList;
    private View view;

    public SliderAdapter(Activity context, ArrayList<Update> updateList) {
        this.mContext = context;
        this.mUpdateList = updateList;
    }

    @Override
    public int getCount() {
        return mUpdateList.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view == object;
    }


    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {

        Update update = mUpdateList.get(position);
        LayoutInflater layoutInflater = (LayoutInflater) mContext.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        if (layoutInflater != null) {
            view = layoutInflater.inflate(R.layout.slider_item, container, false);
            ImageView imageView = view.findViewById(R.id.slider_imageView);
            mContext.runOnUiThread(new Runnable() {
                @Override
                public void run() {
                    Glide.with(mContext)
                            .load(update.getUrl())
                            .placeholder(R.drawable.background_gradient)
                            .into(imageView);
                    ViewPager viewPager = (ViewPager) container;
                    viewPager.addView(view, 0);
                }
            });
        }

        return view;
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        ViewPager viewPager = (ViewPager) container;
        View view = (View) object;
        viewPager.removeView(view);
    }
    public void setItems(ArrayList<Update> updates){
        this.mUpdateList = updates;
    }
}

File: app/src/main/java/in/completecourse/adapter/SliderAdapter.kt
package `in`.completecourse.adapter

import `in`.completecourse.R
import `in`.completecourse.model.Update
import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.fragment.app.FragmentActivity
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.bumptech.glide.Glide
import java.util.*

class SliderAdapter(private val mContext: FragmentActivity?, private var mUpdateList: ArrayList<Update>?) : PagerAdapter() {
    private var view: View? = null

    override fun getCount(): Int {
        return mUpdateList!!.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val update = mUpdateList?.get(position)
        val layoutInflater = mContext?.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater
        view = layoutInflater.inflate(R.layout.slider_item, container, false)
        val imageView = view!!.findViewById<ImageView>(R.id.slider_imageView)
        mContext.runOnUiThread {
            Glide.with(view!!.context)
                    .load(update?.url)
                    .placeholder(R.drawable.background_gradient)
                    .into(imageView)
            val viewPager = container as ViewPager
            viewPager.addView(view, 0)
        }
        return view!!
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val viewPager = container as ViewPager
        val view = `object` as View
        viewPager.removeView(view)
    }

    fun setItems(updates: ArrayList<Update>?) {
        mUpdateList = updates
    }

}